circleci/circleci-docs
# Circle 13442 add webpack build

# Description
- Add a basic webpack build so that we can include other modules
- Create html snippet to load built js file
- Update circle config so that new assets are persisted

# Reasons
- We want to have more visibility into errors thrown from this part of the site

# Considerations 
- ~We considered the fact that browsers will cache files by name. This would cause it so that the users receive a cached version of our js files, meaning they could be running outdated code~.
    - ~To get around that, we wanted to have a file hash in the url~
    - ~It seemed difficult to get Jekyll to include a dynamic file name~
    - ~An html snippet produced by Webpack, to be included by Jekyll, seemed like the best path forward 😄~
- We found out that Jekyll does filename hashing in order to avoid this problem, so we don't need to further solve it on our end.

# How to test 
- `$ npm run webpack-dev`
- Start server
- Go to <http://localhost:4000/docs/>
- Open up the console
- run `Cookies`, it should be defined 👍

# Follow ups
- This does not include webpack in our development build process. And immediate follow up would be to set up sometime of task automation (rake or npm) that runs webpack in watch mode and that will get kicked off as part of running the server
